internal/service/users/people.go

```diff
@@ -36,9 +44,70 @@ func (s *Service) GetAllPeople() (openapi.People, error) {
 			Agreements: openapi.ProfileAgreements{
 				ConfirmedAgreements: agreements,
 			},
-			Roles: roles,
+			TrainingRecord: training,
+			Roles:          roles,
 		}
 		people = append(people, person)
 	}
 	return people, nil
 }
+
+func (s *Service) GetPerson(id string) (types.User, error) {
+	person := types.User{}
+	result := s.db.Where("id = ?", id).
+		First(&person)
+	if result.Error != nil {
+		return person, result.Error
+	}
+	return person, nil
+}
+
+func (s *Service) GetPersonTrainingRecords(user types.User) (openapi.PersonTrainingRecords, error) {
+	// get the user training records, extract the relevant data and put it in a TrainingRecord format (only date and kind)
+	record := types.UserTrainingRecord{
+		UserID: user.ID,
+	}
+	var fetchedTrainingRecords []types.UserTrainingRecord
+	result := s.db.Where(&record).Find(&fetchedTrainingRecords)
+	if result.Error != nil {
+		return openapi.PersonTrainingRecords{}, result.Error
+	}
+
+	var trainingRecords []openapi.TrainingRecord
+
+	for _, tr := range fetchedTrainingRecords {
+		trainingKind := openapi.TrainingKind(tr.Kind)
+
+		var formattedTime *string
+		if !tr.CompletedAt.IsZero() {
+			formatted := tr.CompletedAt.Format(time.RFC3339)
+			formattedTime = &formatted
+		}
+
+		apiTrainingRecord := openapi.TrainingRecord{
+			Kind:        trainingKind,
+			CompletedAt: formattedTime,
+		}
+
+		trainingRecords = append(trainingRecords, apiTrainingRecord)
+	}
+
+	return trainingRecords, nil
+}
+
+func (s *Service) SetTrainingValidity(user types.User, trainingkind types.TrainingKind, date string) error {
+
+	confirmationTime, err := time.Parse(time.RFC3339, date)
+	if err != nil {
+		return err
+	}
+	switch trainingkind {
+	case types.TrainingKindNHSD:
+
+		response := s.createNHSDTrainingRecord(user, confirmationTime)
+
+		return response
+	}
```

Comment:
Whats the intended use of the switch here?

Reply:
Since we don't have multiple types yet, I'd avoid adding code in anticipation. Keep it simple until you need to add extra logic. Think YAGNI as the default.

Reply:
given the training kind is part of the request I vote we keep the switch (so when/if other types are added you don't get the wrong behaviour), but have add a `default: panic("unsupported training kind")`

Reply:
If its 2 for keeping it and 1 against then it looks like I'm outvoted 😁